### PR TITLE
Route ids

### LINKED ISSUE #53: Dojo 7: Automatically create an id in the routes.ts or routes.tsx module
In dojo 7 there is a requirement to provide an `id` in the routing configuration - for applications migrating from dojo 6 to dojo7 it is safe to create an id with the same value as the `outlet` property.

### PR DESCRIPTION
**Pull request by maier49**
Adds route IDs with the same value as the outlet to `routes.ts` or `routes.tsx`

Builds on #54 

Resolves #53

### CODE REVIEW

File: src/v7/transforms/route-ids.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { getLineEndings } from '../../util';
+
+export default function(file: any, api: any) {
+	let quote: string | undefined;
+
+	if (file.path && /^routes\.ts(x)?$/.test(file.path)) {
```
Comment (agubler) [inline src/v7/transforms/route-ids.ts:6]: Will this do `src/routes`, is the `src` part implied?

Comment (maier49) [reply src/v7/transforms/route-ids.ts]: Nope, that's a mistake, which also made me realize I hadn't tested this one against an actual app. Removed the caret and tested all three combined against a fresh `dojo create app` 👍

File: src/v7/transforms/route-ids.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import { getLineEndings } from '../../util';
+
+export default function(file: any, api: any) {
+	let quote: string | undefined;
+
+	if (file.path && /routes\.ts(x)?$/.test(file.path)) {
```
Comment (agubler) [inline src/v7/transforms/route-ids.ts:6]: Do you think we should only do this for `src/routes.(ts|tsx)` not for any other route files in the code base? Also would the updated regex also process files like `src/mailroutes.tsx`?

**Comment by agubler** [comment]
Another nice to have for this would be to detect if the application is using routing? and if it is but no `src/routes.(ts|tsx)` file is found to output a warning?

File: src/v7/transforms/route-ids.ts (hunk 2 above)
Comment (maier49) [reply src/v7/transforms/route-ids.ts]: Yes I do, yes it would 🤦‍♂

**Comment by maier49** [comment]
@agubler Ok I'll work on adding that detection/warning

**Comment by maier49** [comment]
@agubler Ok so now if we see a file using `registerRouterInjector` and none of the imports will match the route regex (determined by resolving the import relative to the directory of the current file) then it logs out a warning.

**Pull request merged**